src/CwpSearchResult.php

```diff
@@ -61,20 +62,34 @@ class CwpSearchResult extends ViewableData
     public function __construct($terms = '', ArrayData $results = null)
     {
         $this->query = $terms;
+        $indexableService = $this->getIndexableService();
         if ($results) {
             // Clean up the results.
             $matches = $results->Matches;
             foreach ($matches as $result) {
-                if (!$result->canView()) {
-                    $matches->remove($result);
+                if ($indexableService) {
+                    if (!$indexableService->isIndexable($result)) {
+                        $matches->remove($result);
+                    }
+                } else {
+                    if (!$result->canView()) {
+                        $matches->remove($result);
+                    }
                 }
             }
-
             $this->matches = $matches;
             $this->suggestion = $results->SuggestionNice;
         }
     }
 
+    private function getIndexableService()
```

Comment:
Probably better off not doing this, just use ^3.7 in composer.json for fulltextsearch (or whatever the correct version is) and that will always have IndexableService